extensions/resource-deployment/src/ui/resourceTypeWizard.ts

```diff
@@ -35,21 +40,30 @@ export class ResourceTypeWizard {
 		this._useGenerateScriptButton = value;
 	}
 
-	//TODO: eventually only resourceType will be passed. For now, we are passing both the resourceType and provider
 	constructor(
 		public resourceType: ResourceType,
-		public provider: DeploymentProvider,
 		public _kubeService: IKubeService,
 		public azdataService: IAzdataService,
 		public notebookService: INotebookService,
 		public toolsService: IToolsService,
-		public platformService: IPlatformService) {
-		this.wizardObject = azdata.window.createWizard(resourceType.displayName, resourceType.name, 'wide');
-		this.model = this.getResourceProviderModel()!;
+		public platformService: IPlatformService,
+		public resourceTypeService: ResourceTypeService) {
+
+		if (resourceType.options) {
+			this.provider = this.resourceType.getProvider(resourceType.options.map(option => { return { option: option.name, value: option.values[0].name }; }))!;
+		} else {
+			this.provider = this.resourceType.providers[0];
+		}
 	}
 
 
-	public getResourceProviderModel(): ResourceTypeModel | undefined {
+	public getResourceProviderModel(): ResourceTypeModel {
+		/**
+		 * Currently changing wizard tiltes and pages does not work without closing and reopening the wizard. (it makes the changes to objects but visually everything remains the same).
+		 * Also, the done button listener gets broken when we close and reopen the same dialog
+		 * For these reasons, I am creating a new wizard every time user changes the options.
+		 */
+		this.wizardObject = azdata.window.createWizard(this.resourceType.displayName, this.resourceType.name, 'wide');
```

Comment:
This still looks like the wrong place to create the wizardObject because whatever the wizardObject depends on was already known

Reply:
The wizardObject creation depends only on this.resourceType so it shoudl be constructed in constructor of whereever this.resourceType gets updated.